installed-tests/js/testGObjectInterface.js

```diff
@@ -0,0 +1,323 @@
+// -*- mode: js; indent-tabs-mode: nil -*-
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const GObject = imports.gi.GObject;
+const JSUnit = imports.jsUnit;
+const Lang = imports.lang;
+const Mainloop = imports.mainloop;
+
+const MyInterface = new Lang.Interface({
+    Name: 'MyInterface',
+});
+
+const MyInterfaceGObject = new Lang.Class({
+    Name: 'MyInterfaceGObject',
+    Extends: GObject.Object,
+    Implements: [ MyInterface ],
+
+    _init: function (props={}) {
+        this.parent(props);
+    }
+});
+
+const MyGObjectInterface = new Lang.Interface({
+    Name: 'MyGObjectInterface',
+    GTypeName: 'ArbitraryGTypeName',
+    Requires: [ GObject.Object ],
+    Properties: {
+        'interface-prop': GObject.ParamSpec.string('interface-prop',
+            'Interface property', 'Must be overridden in implementation',
+            GObject.ParamFlags.READABLE,
+            'foobar')
+    },
+    Signals: {
+        'interface-signal': {}
+    },
+
+    requiredG: Lang.Interface.UNIMPLEMENTED,
+    optionalG: function () {
```

Comment:
I probably missed something, but I'm a little confused by the notion of optional functions on interfaces. My understanding is that `GInterface` specifically requires callers to implement all its vfuncs. This appears to me to be something more like an abstract class than an interface a-la `GInterface`. Perhaps the intention was the model ABCs like in Python? I'm not sure.